docs_src/microservices/general/messagebus.md

```diff
@@ -0,0 +1,368 @@
+
+#### Configuration
+
+See [Common Configuration](#common-configuration) section above for the common configuration elements for all implementations.
+
+##### Security Configuration 
+
+| Option     | Default Value      | Description                                                  |
+| ---------- | ------------------ | ------------------------------------------------------------ |
+| AuthMode   | `usernamepassword` | Mode of authentication to use. Values are `none`, `usernamepassword`<br />, `clientcert`, or `cacert` |
+| SecretName | `redisb`           | Secret name used to look up credentials in the service's SecretStore |
+
+##### Additional Configuration
+
+This implementation does not have any additional configuration.
+
+### MQTT 3.1
+
+Robust message bus protocol, which has additional configuration options for robustness and requires an additional MQTT Broker to be running. See [MQTT Spec](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html) for more details on this protocol.
+
+#### Configuration
+
+See [Common Configuration](#common-configuration) section above for the common configuration elements for all implementations.
+
+##### Security Configuration 
+
+| Option     | Default Value | Description                                                  |
+| ---------- | ------------- | ------------------------------------------------------------ |
+| AuthMode   | `none`        | Mode of authentication to use. Values are `none`, `usernamepassword`, `clientcert`, or `cacert`. The MQTT Broker is currently not secured in secure mode. |
+| SecretName | blank         | Secret name used to look up credentials in the service's SecretStore |
+
+##### Additional Configuration
+
+| Option    | Default Value                                     | Description                                                  |
+| -------------- | ------------------------------------------------------------ | -------------- |
+| ClientId       | service key | Unique name of the client connecting to the MQTT broker      |
+| Qos            | `0` | Quality of Service level <br />0: At most once delivery<br />1: At least once delivery<br />2: Exactly once delivery<br />See the [MQTT QOS Spec](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718099) for more details |
+| KeepAlive     | `10s`         | Maximum time interval in seconds that is permitted to elapse between the point at which the client finishes transmitting one control packet and the point it starts sending the next. If exceeded, the broker will close the client connection |
+| Retained       | `false` | If true, Server MUST store the Application Message and its QoS, so that it can be delivered to future subscribers whose subscriptions match its topic name |

```

Comment:
Does the retention only apply to the last message? Is there a way to see the retained flag from the mqtt broker?

Reply:
Just the last message for a given topic per the MQTT documentation here: https://www.hivemq.com/blog/mqtt-essentials-part-8-retained-messages/

**_A retained message is a normal MQTT message with the retained flag set to true. The broker stores the last retained message and the corresponding QoS for that topic._**

Reply:
I'll add that link to the description